### PR TITLE
Make all BooPicklers explicit

### PR DESCRIPTION
**Pull request by cquiroz**
I've noted we don't always explicitly define all boo picklers leading to bigger size. From now on this should fail to compile rather than use the auto generated pickler via macros

### CODE REVIEW

**Comment by cquiroz** [comment]
This changes reduces in about 1% the js size just by virtue of removing the boopickle macros

File: modules/seqexec/web/shared/src/main/scala/seqexec/web/model/boopickle/ModelBooPicklers.scala [hunk 1]
```diff
@@ -30,13 +31,28 @@ trait ModelBooPicklers extends GemModelBooPicklers {
   // IMPORTANT The order of the picklers is very relevant to the generated size
   // add them with care
   //**********************
```
Comment (tpolecat) [inline modules/seqexec/web/shared/src/main/scala/seqexec/web/model/boopickle/ModelBooPicklers.scala]: Does this comment still apply?

File: modules/seqexec/web/shared/src/main/scala/seqexec/web/model/boopickle/ModelBooPicklers.scala [hunk 2]
```diff
@@ -30,13 +31,28 @@ trait ModelBooPicklers extends GemModelBooPicklers {
...
-  implicit val instrumentPickler = generatePickler[Instrument]
+  implicit val instrumentPickler = transformPickler(
+    (t: Int) =>
+      Instrument.all
+        .find(_.ordinal === t)
+        .getOrElse(throw new RuntimeException("Failed to decode instrument")))(
```
Comment (tpolecat) [inline modules/seqexec/web/shared/src/main/scala/seqexec/web/model/boopickle/ModelBooPicklers.scala]: It seems like we could define `enumPickler[A: Enumerated]` and factor out the repetition here and below. Can we add the ordinal to the error message?

Comment (tpolecat) [reply modules/seqexec/web/shared/src/main/scala/seqexec/web/model/boopickle/ModelBooPicklers.scala]: It also might be worth making a map or array or something so the lookup isn't linear with `.find` … I don't know how much of a bottleneck this is.

Comment (cquiroz) [reply modules/seqexec/web/shared/src/main/scala/seqexec/web/model/boopickle/ModelBooPicklers.scala]: Yeah, though these are not `Enumerated` yet

File: modules/seqexec/web/shared/src/main/scala/seqexec/web/model/boopickle/ModelBooPicklers.scala (hunk 1 above)
Comment (cquiroz) [reply modules/seqexec/web/shared/src/main/scala/seqexec/web/model/boopickle/ModelBooPicklers.scala]: I'll remove it

File: modules/seqexec/web/shared/src/main/scala/seqexec/web/model/boopickle/ModelBooPicklers.scala [hunk 3]
```diff
@@ -26,17 +29,38 @@ import seqexec.model.events._
     "org.wartremover.warts.OptionPartial"
   ))
 trait ModelBooPicklers extends GemModelBooPicklers {
-  //**********************
-  // IMPORTANT The order of the picklers is very relevant to the generated size
-  // add them with care
-  //**********************
-  implicit val instrumentPickler = generatePickler[Instrument]
+  def valuesMap[F[_]: Functor: Foldable, A, B](c: F[A], f: A => B): Map[B, A] =
```
Comment (cquiroz) [inline modules/seqexec/web/shared/src/main/scala/seqexec/web/model/boopickle/ModelBooPicklers.scala]: @tpolecat, does this make sense? Do you know why I can’t do just `valueMap(List, _.ordinal)` when calling it?

Instead I have to declare explicitly the arg type

**Pull request merged**